DEVELOPING.md

```diff
@@ -0,0 +1,126 @@
+# Logstash Development Workflow
```

Comment:
I wouldn't call this top level document the Workflow, as it mainly describes the available tooling and tasks available to the developer
Maybe just "Logstash Development" ?

Then we can add the second main section which describes the processes of:
* creating and submitting a PR
* reviewing and merging a PR
* developing in a feature branch